src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java

```diff
@@ -206,7 +206,9 @@ private static void verifyHasProductionFile(Map<String, List<String>> allTests,
                     // JavadocMetadataScraper and related classes are temporarily hosted in test
                     && !file.getPath().contains("meta")
                     // InlineConfigParser is hosted in test
-                    && !file.getPath().contains("bdd")) {
+                    && !file.getPath().contains("bdd")
+                    // Annotation to suppress invocation of forbidden apis
+                    && !"SuppressForbiddenApi.java".equals(fileName)) {
```

Comment:
Move whole IF expression to new method, proposed name "isTarget".
It will remove problem with Complexity.

Reply:
done